File: libView/src/main/java/org/waynezhou/libView/LibView.java
package org.waynezhou.libView;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

import org.waynezhou.libUtil.reflection.ReflectionException;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

public final class LibView {
    private LibView() {
    }
    
    @SuppressWarnings({"unchecked", "ConstantConditions"})
    @NonNull
    public static <TViewBinding extends ViewBinding> TViewBinding inflate(
            @NonNull LayoutInflater layoutInflater,
            @NonNull Class<TViewBinding> viewBindingType,
            @Nullable ViewGroup container,
            @NonNull Boolean attachToParent
    ) throws ReflectionException {
        try {
            Method inflateMethod = viewBindingType.getMethod(
                    "inflate",
                    LayoutInflater.class,
                    ViewGroup.class,
                    boolean.class
            );
            return (TViewBinding) (inflateMethod.invoke(null, layoutInflater, container, attachToParent));
        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
            throw new ReflectionException(e);
        }
    }
}

File: libViewKt/src/main/java/org/waynezhou/libviewkt/LibView.kt
package org.waynezhou.libviewkt

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import org.waynezhou.libutilkt.reflection.ReflectionException
import java.lang.reflect.InvocationTargetException

object LibView {
    @Throws(ReflectionException::class)
    @Suppress("UNCHECKED_CAST")
    fun <TViewBinding : ViewBinding> LayoutInflater.inflate(
        viewBindingType: Class<TViewBinding>,
        container: ViewGroup?,
        attachToParent: Boolean
    ): TViewBinding {
        return try {
            val inflateMethod = viewBindingType.getMethod(
                "inflate",
                LayoutInflater::class.java,
                ViewGroup::class.java,
                Boolean::class.javaPrimitiveType
            )
            inflateMethod.invoke(null, this, container, attachToParent) as TViewBinding
        } catch (e: IllegalAccessException) {
            throw ReflectionException(e)
        } catch (e: InvocationTargetException) {
            throw ReflectionException(e)
        } catch (e: NoSuchMethodException) {
            throw ReflectionException(e)
        }
    }

}